email/ncmec_test.go

```diff
@@ -0,0 +1,163 @@
+package email
+
+import (
+	"strings"
+	"testing"
+	"time"
+)
+
+// TestNCMECClient contains a couple of unit tests that verify the functionality
+// of the NCMEC client.
+func TestNCMECClient(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	// os.Setenv("NCMEC_USERNAME", "[ENTER_USERNAME]")
+	// os.Setenv("NCMEC_PASSWORD", "[ENTER_PASSWORD]")
+	// os.Setenv("NCMEC_DEBUG", "true")
+
+	// load credentials from env
+	creds, err := LoadNCMECCredentials()
+	if err != nil {
+		t.Log("NCMEC credentials not found, skipping...")
+		t.SkipNow()
+	}
+
+	// ensure it's a debug client
+	if !creds.Debug {
+		t.Fatal("never run in production mode")
+	}
+
+	// instantiate the client
+	client := NewNCMECClient(creds)
+
+	tests := []struct {
+		name string
+		test func(t *testing.T)
+	}{
+		{
+			name: "finishReport",
+			test: func(t *testing.T) { testFinishReport(t, client) },
+		},
+		{
+			name: "openReport",
+			test: func(t *testing.T) { testOpenReport(t, client) },
+		},
+		{
+			name: "status",
+			test: func(t *testing.T) { testStatus(t, client) },
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, test.test)
+	}
+}
+
+// testFinishReport is a unit test that verifies whether we can finish a report
+func testFinishReport(t *testing.T, c *NCMECClient) {
+	// open a report
+	now := time.Now().Add(-time.Hour)
+	report := newTestReport(now)
+	res, err := c.openReport(report)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// finish the report
+	ress, err := c.finishReport(res.ReportId)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// assert the report was finished
+	if ress.ResponseCode != 0 {
+		t.Fatalf("unexpected response code, %v != 0", ress.ResponseCode)
+	}
+	if ress.ReportId != res.ReportId {
+		t.Fatalf("unexpected report id, %v != %v", ress.ReportId, res.ReportId)
+	}
+}
+
+// testOpenReport is a unit test that verifies we can open a report with NCMEC
+func testOpenReport(t *testing.T, c *NCMECClient) {
+	// open a report
+	now := time.Now().Add(time.Hour)
+	report := newTestReport(now)
+	res, err := c.openReport(report)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// assert the response, we expect it to have failed because the invalid
+	// (future) date
+	if res.ResponseCode != 4100 {
```

Comment:
The seem to be using custom response codes. It would be nice to export those to constants with descriptive names. Otherwise, anyone trying to maintain this code will need to look up their help pages and go through all the research you did.

Reply:
done 👍